### PR TITLE
Add attribute fields style filter

### PR DESCRIPTION
**Pull request by Waguramu**
**Goal**: allow users to filter attribute styles based on field values.

**Scope**:

* rule.h
* rule.cpp
* visialization.cpp
* SourceDataPanelComponent (#171)

**Deliverables**:
- A style property to match a simfil query against attributes' fields.

**How to test**:
1. Build as usual.
2. Serve.
3. Set `attribute-mask` property on a style with `aspect: attribute` property.

**Notes**:
* This PR also includes fix for #171

### CODE REVIEW

**Comment by Waguramu** [comment]
- [ ] Investigate styles not reloading on checkboxes' state change
- [ ] Investigate selecting lane groups from feature search results jumps to wrong location

File: README.md [hunk 1]
```diff
@@ -99,6 +99,7 @@ Each rule within the YAML `rules` array can have the following fields:
...
 | `relation-merge-twoway`       | Specifies whether bidirectional relations should be followed and merged.                                                                                                                                    | Boolean                                                    | `true`, `false`                                    |
 | `attribute-type`              | A regular expression to match against an attribute type.                                                                                                                                                    | String                                                     | `SPEED_LIMIT_.*`                                   |
+| `attribute-mask`              | A regular expression to match against a value of the attribute's field.                                                                                                                                     | String                                                     | `attributeValue.speedLimitKmh > 100`               |
```
Comment (josephbirkner) [inline README.md:102]: 1. Please change to `attribute-filter`
2. Check the description :)

File: libs/core/include/erdblick/rule.h [hunk 2]
```diff
@@ -78,6 +78,7 @@ class FeatureStyleRule
     [[nodiscard]] bool relationMergeTwoWay() const;
 
     [[nodiscard]] std::optional<std::regex> const& attributeType() const;
+    [[nodiscard]] std::optional<std::string> const& attributeMask() const;
```
Comment (josephbirkner) [inline libs/core/include/erdblick/rule.h:81]: Please rename to `attributeFilter`

File: libs/core/src/visualization.cpp [hunk 3]
```diff
@@ -281,6 +281,52 @@ void FeatureLayerVisualization::addFeature(
             }
             // Iterate over all the layer's attributes.
             layer->forEachAttribute([&, this](auto&& attr){
+                // Check if the attribute's values match the attribute mask for the rule.
+                if (auto const& attrMask = rule.attributeMask()) {
```
Comment (josephbirkner) [inline libs/core/src/visualization.cpp:285]: Just run this inside of `addAttribute` then we can get rid of the simfil context setup duplication.

File: libs/core/src/visualization.cpp [hunk 4]
```diff
@@ -281,6 +281,52 @@ void FeatureLayerVisualization::addFeature(
...
+                                        return evaluateExpression(str, attrEvaluationContext);
+                                    } catch (const std::exception& e) {
+                                        std::cerr << "Error evaluating attribute expression: " << e.what() << std::endl;
+                                        return simfil::Value::null();
+                                    }
+                                }};
+
+                            try {
+                                auto result = boundEvalFun.eval_(*attrMask);
+                                if ((result.isa(simfil::ValueType::Bool) && !(result.as<simfil::ValueType::Bool>())) ||
```
Comment (josephbirkner) [inline libs/core/src/visualization.cpp:315]: Use `result.template as<ValueType::Bool>()` to fix the compile error.

**Pull request merged**